src/server/game/Spells/SpellEffects.cpp

```diff
@@ -1600,7 +1600,9 @@ void Spell::EffectPersistentAA()
         return;
 
     ASSERT(_dynObjAura->GetDynobjOwner());
-    _dynObjAura->_ApplyEffectForTargets(effectInfo->EffectIndex);
+    for (size_t i = 0; i < m_spellInfo->GetEffects().size(); ++i)
+        if (!m_spellInfo->GetEffect(SpellEffIndex(i)).IsEffect(SPELL_EFFECT_NONE))
```

Comment:
You should only be calling _ApplyEffectForTargets for SPELL_EFFECT_PERSISTENT_AREA_AURA effects, not any non-empty effect

Reply:
like this ?